lib/collection/src/collection_manager/holders/proxy_segment.rs

```diff
@@ -381,8 +383,31 @@ impl SegmentEntry for ProxySegment {
         true
     }
 
-    fn flush(&self) -> OperationResult<u64> {
-        Ok(self.wrapped_segment.get().read().version())
+    fn flush(&self) -> OperationResult<SeqNumberType> {
```

Comment:
does it make sense to have maybe a `log` statement in one of the `if` branch so we can see what's happening?

Reply:
Flush is rather frequent operation, we would need some over-complicated logic to prevent flood in logs